src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -1120,19 +1122,51 @@ void SpatialPooler::updateDutyCyclesHelper_(vector<Real>& dutyCycles,
 
 void SpatialPooler::updateBoostFactors_()
 {
-  for (UInt i = 0; i < numColumns_; i++)
+
+  if (maxBoost_ > 1.0)
   {
-    if (minActiveDutyCycles_[i] <= 0)
+    vector<Real> targetDensity(numColumns_, 0);
+    if (globalInhibition_)
     {
-      continue;
+      Real density = localAreaDensity_;
+      if (numActiveColumnsPerInhArea_ > 0)
+      {
+        UInt inhibitionArea = pow((Real) (2 * inhibitionRadius_ + 1),
+                                  (Real) columnDimensions_.size());
+        inhibitionArea = min(inhibitionArea, numColumns_);
+        density = ((Real) numActiveColumnsPerInhArea_) / inhibitionArea;
+        density = min(density, (Real) 0.5);
+      }
+
+      for (UInt i = 0; i < numColumns_; ++i)
+      {
+        targetDensity[i] = density;
+      }
     }
-    if (activeDutyCycles_[i] > minActiveDutyCycles_[i])
+    else
     {
-      boostFactors_[i] = 1.0;
-      continue;
+      for (UInt i = 0; i < numColumns_; ++i)
+      {
+        UInt numNeighbors = 0;
+        Real localActivityDensity = 0;
+        for (UInt neighbor : WrappingNeighborhood(i, inhibitionRadius_,
```

Comment:
Should this obey the `wrapAround_` parameter? i.e. I think there should be one case that uses the `Neighborhood` and the other that uses the `WrappingNeighborhood`.